lib/cocoapods/xcode/linkage_analyzer.rb

```diff
@@ -0,0 +1,22 @@
+require 'macho'
+
+module Pod
+  module Xcode
+    class LinkageAnalyzer
+      # @param  [Pathname] binary
+      #         The file to be checked for being a dynamic Mach-O binary.
+      #
+      # @return [Boolean] Whether `binary` can be dynamically linked.
+      #
+      def self.dynamic_binary?(binary)
+        @@cached_dynamic_binary_results ||= {}
```

Comment:
please don't use class variables. additionally, should this be a cache object, so we don't have global mutable state floating around?

Reply:
this was copied/pasted agreed with overall comment though.

Reply:
ah yeah responded above but this was ported from FileAccessor. I'm open to making a singleton here if that makes more sense